tool/cmd/kitex/versions/dependencies.go

```diff
@@ -0,0 +1,325 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package versions
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/cloudwego/kitex"
+	"github.com/cloudwego/kitex/tool/internal_pkg/log"
+)
+
+const (
+	CompatibilityCheckExitCode = 3
+)
+
+var dm *dependencyManager
+
+func init() {
+	dm = &dependencyManager{}
+}
+
+// RegisterMinDepVersion registers Minimal Dependency Version to be checked
+func RegisterMinDepVersion(depVer *MinDepVersion) error {
+	return dm.Register(depVer)
+}
+
+// CheckDependency is responsible for checking the compatibility of dependency
+// and return the checking results.
+// After calling CheckDependencies(), you should check if the result is Nil:
+//
+//	res := CheckDependency()
+//	if res != nil {
+//	}
+//
+// If res is not nil, pls check the related error with `Err()`:
+//
+//		if err := res.Err(); err != nil {
+//		    switch {
+//	        case errors.Is(err, ErrGoCmdNotFound):
+//	        case errors.Is(err, ErrGoModNotFound):
+//		    case errors.Is(err, ErrDependencyNotFound):
+//		    case errors.Is(err, ErrDependencyVersionNotSemantic):
+//		    case errors.Is(err, ErrDependencyReplacedWithLocalRepo):
+//		    case errors.Is(err, ErrDependencyVersionNotCompatible):
+//		    }
+//		}
+//
+// Then you can get the version in go.mod with `GoModVersion()` and retrieve the MinDepVersion
+// information with `MinDepVersion()`.
+func CheckDependency() *CheckResult {
+	return dm.CheckDependency()
+}
+
+// DefaultCheckDependencyAndProcess provided default processing procedure to parse
+// CheckResult and prompt users
+func DefaultCheckDependencyAndProcess() error {
+	cr := CheckDependency()
+	if cr == nil {
+		return nil
+	}
+	res, shouldExit := defaultParseCheckResult(cr)
+	if res != "" {
+		log.Warn(res)
+	}
+	if shouldExit {
+		return errors.New("kitex cmd tool dependency compatibility check failed")
+	}
+	return nil
+}
+
+func defaultParseCheckResult(cr *CheckResult) (prompt string, shouldExit bool) {
+	if cr == nil {
+		return
+	}
+	err := cr.Err()
+	switch {
+	case err == nil:
+		return
+	// ignore ErrGoCmdNotFound
+	case errors.Is(err, ErrGoCmdNotFound):
+	// ignore ErrGoModNotFound
+	case errors.Is(err, ErrGoModNotFound):
+	// ignore ErrDependencyNotFound
+	case errors.Is(err, ErrDependencyNotFound):
+	// ignore ErrDependencyReplacedWithLocalRepo
+	case errors.Is(err, ErrDependencyReplacedWithLocalRepo):
+	// ignore ErrDependencyVersionNotSemantic
+	case errors.Is(err, ErrDependencyVersionNotSemantic):
+	case errors.Is(err, ErrDependencyVersionNotCompatible):
+		prompt = defaultPromptWithCheckResult(cr)
+		shouldExit = true
```

Comment:
seems it can be simplied to 
```go
if errors.Is(err, ErrDependencyVersionNotCompatible) {
    return ...
}
return "", false
```

Reply:
Yes, that's very concise. I was just thinking I'd have to follow the ```CheckDependency``` comments.